File: selection_sort_go/selection_sort.go
package main

import "fmt"

func selectionSort(arr []int) []int {
	for i := 0; i < len(arr); i++ {
		minIndex := i
		for j := i + 1; j < len(arr); j++ {
			if arr[minIndex] > arr[j] {
				minIndex = j
			}
		}
		arr[i], arr[minIndex] = arr[minIndex], arr[i]
	}
	return arr
}

func main() {
	arr := []int{8, 11, 1, 9, 10, 5, 7, 6, 3, 2, 4}
	fmt.Println(selectionSort(arr))
}

File: selection_sort_py/selection_sort.py
def selection_sort(arr):
    for i in range(len(arr)):
        min_index = i
        for j in range(i+1, len(arr)):
            if arr[min_index] > arr[j]:
                min_index = j
        arr[i], arr[min_index] = arr[min_index], arr[i]
    return arr


arr = [8, 11, 1, 9, 10, 5, 7, 6, 3, 2, 4]
print(selection_sort(arr))
